scala/README.md

```diff
@@ -534,22 +610,71 @@ Tips & Tricks
 
     ```scala
     // Value classes
-    case class Email(email: String) extends AnyVal // no extra object allocation at runtime
+    case class Email(email: String) extends AnyVal // No extra object allocation at runtime
 
-    val email = Email("no@one.com") // never gets mixed with regular strings
+    val email = Email("joe@example.com") // Never gets mixed with other strings
```

Comment:
I've been using Scalaz tagged types for almost the same purpose, they should also be zero-overhead but I haven't actually checked yet.